doc/gdb-debugging.md

```diff
@@ -0,0 +1,54 @@
+### GDB Debugging
+This document describes the options available for debugging node using the GNU debugger.
+
+We don't provide the debuginfo package, which contains the symbols required for debugging, as this would make our runtime image about 480MB larger. On systems where root access is allowed then the debuinfo package can simply be installed using yum/rpm, but on others like OpenShift getting a pod to run with root access can be somewhat tricky. 
+
+### OpenShift
+When running an application on OpenShift the pod will not have root access. It is possible to configure such access but that means configuring the OpenShift environment. This section provides an alternative to that which allows debugging the application using a remote gdb session. This is done by starting a container in the same cluster with an image with the node runtime and debuginfo, and then connecting to the pod that is the target of the debugging.
+
+#### Start/attach to the running process
+First step is to start the gdbserver on the pod that is the target of the debugging session:
+```
+$ oc rsh nodejs-rest-http-1-nw7hz
+$ gdbserver 172.17.0.6:7777 --attach 23
+Attached; pid = 23
+Listening on port 7777
+```
+In the above case we are creating a new process, but this could also just be the process id of a process running on the pod.
+
+#### Start the debuginfo image
+The next step is to start a container with the debuginfo image in the same cluster.
+
+```console
+$ oc run -i -t nodejs-debuginfo --image=bucharestgold/centos7-s2i-nodejs2-debuginfo:10.x --restart=Never
```

Comment:
is that 2 suppose to be at the end of nodejs?

Reply:
For some reason I was not able to get the image to be updated unless I renamed it, hence the appended 2 on the name. I've fix this. Thanks